src/components/UI/dropdown/index.js

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState, useRef } from 'react';
+import Link from 'next/link';
+
+import styles from './dropdown.module.scss';
+
+const Dropdown = ({
+  isLoggedIn,
+  USER_PROFILE_URL,
+  SIGN_OUT_URL,
+  userData,
+  DEFAULT_AVATAR,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const toggleDropdown = () => setIsOpen(!isOpen);
+  const modalClose = useRef();
+
+  const signOut = () => {
+    fetch(SIGN_OUT_URL, {
+      method: 'GET',
+      credentials: 'include',
+    }).then(() => {
+      window.location.reload();
+    });
+  };
+
+  useEffect(() => {
+    const handler = (event) => {
+      if (!modalClose.current.contains(event.target)) {
+        setIsOpen(false);
+      }
+    };
+    document.addEventListener('mousedown', handler);
+    return () => {
+      document.removeEventListener('mousedown', handler);
+    };
+  });
+
+  return (
+    <div className={styles.userGreet} ref={modalClose}>
+      <div
+        role="button"
+        tabIndex={0}
+        className={styles.dropdownHeader}
+        onClick={toggleDropdown}
+        onKeyDown={toggleDropdown}
```

Comment:
Because of this error.
![image](https://user-images.githubusercontent.com/73482745/192686696-7eaf70c7-d3cc-40d2-ba39-356291c392b8.png)

I found this solution on google and used in our code 
https://stackoverflow.com/questions/48575674/how-to-add-a-keyboard-listener-to-my-onclick-handler

Reply:
okay got it